packages-content-model/roosterjs-content-model-core/lib/corePlugin/LifecyclePlugin.ts

```diff
@@ -0,0 +1,205 @@
+
+const ContentEditableAttributeName = 'contenteditable';
+const DefaultTextColor = '#000000';
+const DefaultBackColor = '#ffffff';
+
+/**
+ * Lifecycle plugin handles editor initialization and disposing
+ */
+class LifecyclePlugin implements PluginWithState<LifecyclePluginState> {
+    private editor: (IStandaloneEditor & IEditor) | null = null;
+    private state: LifecyclePluginState;
+    private initialModel: ContentModelDocument;
+    private initializer: (() => void) | null = null;
+    private disposer: (() => void) | null = null;
+    private adjustColor: () => void;
+
+    /**
+     * Construct a new instance of LifecyclePlugin
+     * @param options The editor options
+     * @param contentDiv The editor content DIV
+     */
+    constructor(options: StandaloneEditorOptions, contentDiv: HTMLDivElement) {
+        this.initialModel =
+            options.initialModel ?? this.createInitModel(options.defaultSegmentFormat);
+
+        // Make the container editable and set its selection styles
+        if (contentDiv.getAttribute(ContentEditableAttributeName) === null) {
+            this.initializer = () => {
+                contentDiv.contentEditable = 'true';
+                contentDiv.style.userSelect = 'text';
+            };
+            this.disposer = () => {
+                contentDiv.style.userSelect = '';
+                contentDiv.removeAttribute(ContentEditableAttributeName);
+            };
+        }
+        this.adjustColor = options.doNotAdjustEditorColor
+            ? () => {}
+            : () => {
+                  this.adjustContainerColor(contentDiv);
+              };
+
+        this.state = {
+            isDarkMode: !!options.inDarkMode,
+            onExternalContentTransform: null,
+            shadowEditFragment: null,
+        };
+    }
+
+    /**
+     * Get a friendly name of  this plugin
+     */
+    getName() {
+        return 'Lifecycle';
+    }
+
+    /**
+     * Initialize this plugin. This should only be called from Editor
+     * @param editor Editor instance
+     */
+    initialize(editor: IEditor) {
+        this.editor = editor as IEditor & IStandaloneEditor;
+
+        this.editor.setContentModel(
+            this.initialModel,
+            { ignoreSelection: true },
+            this.editor.isDarkMode() ? this.onInitialNodeCreated : undefined
+        );
+
+        // Set content DIV to be editable
+        this.initializer?.();
+
+        // Set editor background color for dark mode
+        this.adjustColor();
+
+        // Let other plugins know that we are ready
+        this.editor.triggerPluginEvent(PluginEventType.EditorReady, {}, true /*broadcast*/);
+    }
+
+    /**
+     * Dispose this plugin
+     */
+    dispose() {
+        this.editor?.triggerPluginEvent(PluginEventType.BeforeDispose, {}, true /*broadcast*/);
+
+        if (this.disposer) {
+            this.disposer();
+            this.disposer = null;
+            this.initializer = null;
+        }
+
```

Comment:
Just curious, do we need to dispose the initialModel and the state?

Reply:
state only has some simple boolean values, so there is nothing to dispose.
For initialModel, although it is possible large, but it is still just a JSON object without any external links, so it should be fine not dispose it. But I think I can overwrite it using an empty model to make sure cutting off all the links.